smart_open/s3.py

```diff
@@ -80,10 +80,13 @@ def open(bucket_id, key_id, mode, **kwargs):
 
 class RawReader(object):
     """Read an S3 object."""
-    def __init__(self, s3_object):
+    def __init__(self, s3_object, version_id=None):
         self.position = 0
         self._object = s3_object
-        self._body = s3_object.get()['Body']
+        if version_id:
```

Comment:
Do you really need to check if version_id is None? I you explicitly do `s3_object.get(VersionId=None)`, what is the result?

Reply:
Hi, boto validate the parameters using the keys in kwargs, so i i pass VersionId=None it will raise a ParamValidationError:
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter VersionId, value: None, type: <type 'NoneType'>, valid types: <type 'basestring'>

Reply:
Interesting. It's a keyword argument, so it must have a default value. Can you figure out what that default value is?

Reply:
All these parameters are in the kwargs dict, so they don't have a default value.
Boto3 are created automatically using a json definition (see boto3 service.py and json definitions in botocore/data)
Before making a network request to AWS api all parameters are validated and serialized.

Reply:
OK, I see. Can you please make a function:

```python
def _get(s3_object, version=None, **kwargs):
  if version is not None:
      kwargs['Version'] = version
  return s3_object.get(**kwargs)
```

and then use it in your code instead of the conditionals?

Right now, we've checking for "version is not None" in too many places, it's repetitive, error prone and asking for trouble.